### PR TITLE
Custom view docs changes

### PR DESCRIPTION
**Pull request by Anshuman71**
This PR is the suggestions from the tech writers for #3289.

## Content

- General
    - [What’s a custom view](
https://appkit-cv-pr-3321.commercetools.vercel.app/custom-views/what-is-a-custom-view)
    - [Getting started](
https://appkit-cv-pr-3321.commercetools.vercel.app/custom-views/getting-started)
- Development
    - [Page content layout](https://appkit-cv-pr-3321.commercetools.vercel.app/custom-views/development/page-content-layouts)
    - Testing
        - [Testing the Custom View](
https://appkit-cv-pr-3321.commercetools.vercel.app/custom-views/development/testing#testing-the-custom-view)
    - Permissions
        - [Defining constants](
https://appkit-cv-pr-3321.commercetools.vercel.app/custom-views/development/permissions#defining-constants)
- API Reference
    - [Custom View config](
https://appkit-cv-pr-3321.commercetools.vercel.app/custom-views/api-reference/custom-view-config)
    - Application-shell
        - [Components](
https://appkit-cv-pr-3321.commercetools.vercel.app/custom-views/api-reference/commercetools-frontend-application-shell#components)
        - [Test utils](
https://appkit-cv-pr-3321.commercetools.vercel.app/custom-views/api-reference/commercetools-frontend-application-shell#test-utils)
    - Application-shell-connectors
        - [Custom View context
](
https://appkit-cv-pr-3321.commercetools.vercel.app/custom-views/api-reference/commercetools-frontend-application-shell-connectors#custom-view-context)
    - [Application components](https://appkit-cv-pr-3321.commercetools.vercel.app/custom-views/api-reference/commercetools-frontend-application-components)

## Tasks

- [x] Review and shape initial drafts
- [x] Replace instances of Custom Applications with Custom Views in regular content, across each section of the microsite.
- [x] Coordinate with the Shield team to make appropriate changes to code snippets, code example naming, etc.

### CODE REVIEW

File: website-custom-views/src/content/getting-started.mdx
Comment (emmenko) [inline website-custom-views/src/content/getting-started.mdx:15]: Nit: if you change these things, we should also change them in the Custom Application docs, as they are copy-pasted.

Or leave them and focus only on the Custom View specific content. The general things will likely be revisited anyway when you work on the unified docs.

This might help you to speed up the reviews.

Comment (Anshuman71) [reply website-custom-views/src/content/getting-started.mdx:15]: Thanks, @emmenko. I realized this after three files, so let's keep these changes in these 3 files. It's very prominent here but others are not so much.

File: website-custom-views/src/content/api-reference/commercetools-frontend-application-shell-connectors.mdx
Comment (Anshuman71) [inline website-custom-views/src/content/api-reference/commercetools-frontend-application-shell-connectors.mdx:77]: @emmenko I did `js` as it adds highlighting.

File: website-custom-views/src/content/api-reference/commercetools-frontend-application-shell-connectors.mdx
Comment (Anshuman71) [inline website-custom-views/src/content/api-reference/commercetools-frontend-application-shell-connectors.mdx:172]: @emmenko please confirm whether this is part of the Context or not.

File: website-custom-views/src/content/api-reference/commercetools-frontend-application-shell.mdx
Comment (Anshuman71) [inline website-custom-views/src/content/api-reference/commercetools-frontend-application-shell.mdx:452]: @emmenko please double check if `applicationContext` is used in Custom Views. I matched the values available in the `CustomViewContext` and they looked fine.

File: website-custom-views/src/content/api-reference/commercetools-frontend-application-shell.mdx
Comment (Anshuman71) [inline website-custom-views/src/content/api-reference/commercetools-frontend-application-shell.mdx:474]: @emmenko please double check if `ApplicationContextProvider` is used in Custom Views. I matched the values available in the CustomViewContext and they looked fine.

File: website-custom-views/src/content/development/permissions.mdx
Comment (Anshuman71) [inline website-custom-views/src/content/development/permissions.mdx:26]: @emmenko I removed this info from all places because the version `21.21.0` is specific to Custom Applications. 

We can keep the of the content, if you think it's still applicable.

File: website-custom-views/src/content/concepts/integrate-with-your-own-api.mdx
Comment (emmenko) [inline website-custom-views/src/content/concepts/integrate-with-your-own-api.mdx:71]:
```suggestion
* **permissions**: Includes the list of applied permissions for the logged-in user. The permission values are formatted as `can<permission-name>`, for example `canView`.
```

File: website-custom-views/src/content/development/forms.mdx
Comment (emmenko) [inline website-custom-views/src/content/development/forms.mdx:432]: Was this removed on purpose?

File: website-custom-views/src/content/development/permissions.mdx
Comment (emmenko) [inline website-custom-views/src/content/development/permissions.mdx:41]: These values should be specified as string literals so that TS can correctly infer and define the mapping names.

File: website-custom-views/src/content/development/ui-components.mdx
Comment (emmenko) [inline website-custom-views/src/content/development/ui-components.mdx:18]: Why was this removed?

File: website-custom-views/src/content/help-needed.mdx
Comment (emmenko) [inline website-custom-views/src/content/help-needed.mdx:43]: Let's keep this, you can omit the sentence with the version

**Review by emmenko — COMMENTED** [review]
Thanks for the improvements!

There a couple of things to adjust / revert from my perspective, but otherwise looks good 🙌

File: website-custom-views/src/content/api-reference/cli.mdx
Comment (Anshuman71) [reply website-custom-views/src/content/api-reference/cli.mdx]: @emmenko I've fixed all these in [a18827e](https://github.com/commercetools/merchant-center-application-kit/pull/3321/commits/a18827ed47c492bb0e5dfa6c8d47c886138af16c)

File: website-custom-views/src/content/api-reference/cli.mdx
Comment (Anshuman71) [reply website-custom-views/src/content/api-reference/cli.mdx]: Fixed in [a18827e](https://github.com/commercetools/merchant-center-application-kit/pull/3321/commits/a18827ed47c492bb0e5dfa6c8d47c886138af16c)

File: website-custom-views/src/content/api-reference/commercetools-frontend-application-shell-connectors.mdx
Comment (Anshuman71) [reply website-custom-views/src/content/api-reference/commercetools-frontend-application-shell-connectors.mdx]: Fixed in [a18827e](https://github.com/commercetools/merchant-center-application-kit/pull/3321/commits/a18827ed47c492bb0e5dfa6c8d47c886138af16c)

File: website-custom-views/src/content/api-reference/commercetools-frontend-application-shell-connectors.mdx
Comment (Anshuman71) [reply website-custom-views/src/content/api-reference/commercetools-frontend-application-shell-connectors.mdx]: Fixed in [a18827e](https://github.com/commercetools/merchant-center-application-kit/pull/3321/commits/a18827ed47c492bb0e5dfa6c8d47c886138af16c)

File: website-custom-views/src/content/development/ui-components.mdx
Comment (Anshuman71) [reply website-custom-views/src/content/development/ui-components.mdx:18]: Hi @emmenko, I removed it because this page is no longer on the Custom Views website, but it's on Custom Applications, so I thought it was not relevant. Can you confirm if we should put the page as well as this Info box in the Custom Views docs?

Comment (emmenko) [reply website-custom-views/src/content/development/ui-components.mdx:18]: Hmm @CarlosCortizasCT shouldn't we have the `page-content-layouts` in the Custom Views docs as well?

File: website-custom-views/src/content/help-needed.mdx
Comment (Anshuman71) [reply website-custom-views/src/content/help-needed.mdx]: Applied in fb7a811b30d8fb02e7b111509af229d054724206

File: website-custom-views/src/content/development/ui-components.mdx
Comment (CarlosCortizasCT) [reply website-custom-views/src/content/development/ui-components.mdx:18]: The idea is that users wouldn't be using these components in Custom Views as we would like for them to always use the whole available horizontal space.

We can have a discussion about this if you want to.

File: website-custom-views/src/content/development/permissions.mdx
Comment (Anshuman71) [reply website-custom-views/src/content/development/permissions.mdx]: Applied in fb7a811b30d8fb02e7b111509af229d054724206

File: website-custom-views/src/content/api-reference/custom-view-config.mdx
Comment (Anshuman71) [reply website-custom-views/src/content/api-reference/custom-view-config.mdx]: Reverted the commit: 46cee48b702614e01b30f84ec26a9533094936a3

File: website-custom-views/src/content/development/forms.mdx
Comment (Anshuman71) [reply website-custom-views/src/content/development/forms.mdx:432]: Hi @emmenko, I removed it because the [`@application-components`](https://docs.commercetools.com/custom-applications/api-reference/commercetools-frontend-application-components#formmodalpage) page is available in the Custom Application docs but not in Custom Views docs. So I thought this doesn't apply here. If this is not the case, should we bring the (/api-reference/commercetools-frontend-application-components) page and this section back?

Comment (CarlosCortizasCT) [reply website-custom-views/src/content/development/forms.mdx:432]: I think this is because we don't want to have nested pages in Custom Views panels.

Comment (emmenko) [reply website-custom-views/src/content/development/forms.mdx:432]: Thanks @CarlosCortizasCT for clarifying this.

As a follow up question: does it mean that Custom Views shouldn't use different routes?

Comment (CarlosCortizasCT) [reply website-custom-views/src/content/development/forms.mdx:432]: I'm not sure we have though is such a use case until last week when someone raised this potential requirement.

Currently we support routing and navigation could be implemented but we don't expect modals to be used.

@mmaltsev-ct do you think this is something we must discuss about for the MVP?

**Review by emmenko — APPROVED** [review]
Good to go from my side 💯

File: website-custom-views/src/content/development/ui-components.mdx
Comment (CarlosCortizasCT) [reply website-custom-views/src/content/development/ui-components.mdx:18]: @Anshuman71 after further discussion with UI/UX designer, we think we need to keep this content in the docs also for Custom Views.

Comment (Anshuman71) [reply website-custom-views/src/content/development/ui-components.mdx:18]: Thanks @CarlosCortizasCT. Then I'll add both the page and the info box back.

Comment (Anshuman71) [reply website-custom-views/src/content/development/ui-components.mdx:18]: Applied in fdf431b7ec79f20e56179d00922c1763a57f60bb

File: website-custom-views/src/content/api-reference/commercetools-frontend-application-components.mdx
Comment (CarlosCortizasCT) [inline website-custom-views/src/content/api-reference/commercetools-frontend-application-components.mdx:1415]:
```suggestion
Renders the children in a [full page content layout](/development/page-content-layouts#full).
```

File: website-custom-views/src/content/development/forms.mdx
Comment (Anshuman71) [reply website-custom-views/src/content/development/forms.mdx:432]: @CarlosCortizasCT, no more changes required here right?

**Comment by CarlosCortizasCT** [comment]
@Anshuman71 what do you think we should do regarding the errors raised because using the word `native` in the context of `native applications`?

image

**Comment by Anshuman71** [comment]
Thanks, @CarlosCortizasCT, I’ve fixed all the breaking things. Let's wait for @basebander's review before merging this one.

**Review by CarlosCortizasCT — APPROVED** [review]
Looks good to me 💯 

Thanks a lot, Anshuman 🙇

File: website-custom-views/src/content/development/forms.mdx
Comment (CarlosCortizasCT) [reply website-custom-views/src/content/development/forms.mdx:432]: No, thanks 👍

File: website-custom-views/src/content/api-reference/commercetools-frontend-application-shell.mdx
Comment (basebander) [inline website-custom-views/src/content/api-reference/commercetools-frontend-application-shell.mdx:52]:
```suggestion
The `<CustomViewShell>` pre-configures the Custom View entry point routes. In addition, the entry point route is protected by the basic View permission check. This means that a user without permission from your Custom View won't be able to access the Custom View route.
```

File: website-custom-views/src/content/api-reference/custom-view-config.mdx
Comment (basebander) [inline website-custom-views/src/content/api-reference/custom-view-config.mdx:471]:
```suggestion
Since Custom Views are rendered in the context of built-in Merchant Center applications, you need to define the specific views where the Custom View should be rendered.
```

File: website-custom-views/src/content/api-reference/custom-view-config.mdx
Comment (basebander) [inline website-custom-views/src/content/api-reference/custom-view-config.mdx:552]:
```suggestion
To make the Custom View config reusable across different environments, you can use references to environment variables within the config file.
```

File: website-custom-views/src/content/api-reference/custom-webpack-config.mdx
Comment (basebander) [inline website-custom-views/src/content/api-reference/custom-webpack-config.mdx:43]: In Custom Applications it is singular, in Custom Views, plural. Intentional?

File: website-custom-views/src/content/concepts/oauth-scopes-and-user-permissions.mdx
Comment (basebander) [inline website-custom-views/src/content/concepts/oauth-scopes-and-user-permissions.mdx:33]:
```suggestion
However, you might need more granular access control to fulfill specific business requirements. Suppose your Custom View manages products, discounts, and orders, and you want a group of users to only manage products and discounts while another group handles orders.
```
html?

File: website-custom-views/src/content/concepts/oauth-scopes-and-user-permissions.mdx
Comment (basebander) [inline website-custom-views/src/content/concepts/oauth-scopes-and-user-permissions.mdx:79]:
```suggestion
## Additional permission groups
```
perhaps?

File: website-custom-views/src/content/development/permissions.mdx
Comment (basebander) [inline website-custom-views/src/content/development/permissions.mdx:26]:
```suggestion
## Additional permissions groups
```

If the suggestion doesn't work, please fix the typo

File: website-custom-views/src/content/development/permissions.mdx
Comment (basebander) [inline website-custom-views/src/content/development/permissions.mdx:91]: @Anshuman71 Just curious, how did you know to change this?

Comment (Anshuman71) [reply website-custom-views/src/content/development/permissions.mdx:91]: I changed this just to use the permissions with groups with the `useIsAuthorized` hook because we didn't have any examples of this. I was looking for such an example, so I thought it would make sense to add here.

File: website-custom-views/src/content/development/permissions.mdx
Comment (Anshuman71) [reply website-custom-views/src/content/development/permissions.mdx:26]: Thanks, Ryan.

File: website-custom-views/src/content/api-reference/custom-webpack-config.mdx
Comment (Anshuman71) [reply website-custom-views/src/content/api-reference/custom-webpack-config.mdx:43]: Good catch. I'll make singular.

Comment (Anshuman71) [inline website-custom-views/src/content/api-reference/custom-webpack-config.mdx:43]:
```suggestion
The Custom View CLI will then use the configuration exported from this file instead of the default one.
```

Comment (Anshuman71) [reply website-custom-views/src/content/api-reference/custom-webpack-config.mdx]: Resolved here: https://github.com/commercetools/merchant-center-application-kit/pull/3321/commits/8bf3753972b8474aa5f7f07fe103a988d59b4409

**Review by basebander — APPROVED** [review]

**Pull request merged**
